### ISSUE TITLE
Compiler bug: ranged switch in an inline while loop - segfault

### ISSUE THREAD

**Issue by tgschultz** [ziglang/zig#3375]
```zig
test "ranged switch in inline while" {
    var b = @as(u8, 0);

    comptime var i = 0;
    inline while (i < 2) : (i += 1) {
        switch (b) {
            0...1 => {},
            else => {},
        }
    }
}
```

**Comment by jumpnbrownweasel** [comment]
Here is another compiler segfault where a switch is used inside of an inline while. I assume it's the same bug.

```zig
const std = @import("std");

test "switch inside inline while" {
    const a = [3]u8{ 1, 2, 3 };
    const b = [3]u8{ 2, 2, 3 };
    std.debug.print("{} {}\n", .{ inlineArrayOrder1(a, b), inlineArrayOrder1(b, a) }); // this succeeds
    //std.debug.print("{}\n", .{inlineArrayOrder2(a, b)}); // this segfaults
}

pub fn inlineArrayOrder1(a: [3]u8, b: [3]u8) std.math.Order {
    comptime var i: usize = 0;
    inline while (i < 3) : (i += 1) {
        const o = std.math.order(a[i], b[i]);
        if (o == .lt) return .lt;
        if (o == .gt) return .gt;
    }
    return .eq;
}

pub fn inlineArrayOrder2(a: [3]u8, b: [3]u8) std.math.Order {
    comptime var i: usize = 0;
    inline while (i < 3) : (i += 1) {
        switch (std.math.order(a[i], b[i])) {
            .lt => return .lt,
            .gt => return .gt,
            .eq => {},
        }
    }
    return .eq;
}
```

**Comment by Jarred-Sumner** [comment]
Seeing similar behavior with `inline while`. Removing `inline` fixed the segfault.

Stack trace:
```
Process:               zig [72464]
Path:                  /Users/USER/*/zig
Identifier:            zig
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        ??? [72463]
Responsible:           Terminal [1399]
User ID:               501

Date/Time:             2021-05-18 20:11:56.734 -0700
OS Version:            macOS 11.3.1 (20E241)
Report Version:        12
Bridge OS Version:     5.3 (18P4556)
Anonymous UUID:        D8B7103A-DBFD-785B-D249-260CCDF6447C

Sleep/Wake UUID:       2114DFBA-ADBC-4E8F-95CB-63DCD14930DF

Time Awake Since Boot: 680000 seconds
Time Since Wake:       270000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [72464]

VM Regions Near 0:
--> 
    __TEXT                      1010c0000-106674000    [ 85.7M] r-x/r-x SM=COW  /Users/*

Application Specific Information:
dyld2 mode

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   zig                           	0x00000001047cce6c llvm::BasicBlock::getContext() const + 4
1   zig                           	0x0000000104853a52 llvm::BranchInst::BranchInst(llvm::BasicBlock*, llvm::BasicBlock*, llvm::Value*, llvm::Instruction*) + 38
2   zig                           	0x00000001039d81a0 llvm::BranchInst::Create(llvm::BasicBlock*, llvm::BasicBlock*, llvm::Value*, llvm::Instruction*) + 64
3   zig                           	0x00000001042a0eb9 llvm::IRBuilderBase::CreateCondBr(llvm::Value*, llvm::BasicBlock*, llvm::BasicBlock*, llvm::MDNode*, llvm::MDNode*) + 41
4   zig                           	0x00000001013f3b76 ir_render_instruction(CodeGen*, IrExecutableGen*, IrInstGen*) + 10102
5   zig                           	0x00000001013e29a8 codegen_build_object(CodeGen*) + 10280
6   zig                           	0x00000001014a31a1 zig_stage1_build_object + 2273

Thread 1:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 2:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 3:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 4:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 5:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 6:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 7:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 8:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 9:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 10:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 11:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 12:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 13:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 14:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 15:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 16:
0   libsystem_kernel.dylib        	0x00007fff2034e366 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff201d9c7b _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff201da14d _dispatch_semaphore_wait_slow + 98
3   zig                           	0x000000010111c857 std.Thread.MainFuncs.posixThreadMain + 151
4   libsystem_pthread.dylib       	0x00007fff2037f4a7 thread_start + 15

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x00007f8a08ea2a80  rbx: 0x00007f8a08ea2a80  rcx: 0x00007f8a08ea29e0  rdx: 0x00007f8a08e864d0
  rdi: 0x0000000000000000  rsi: 0x0000000000000000  rbp: 0x00007ffeeeb3a100  rsp: 0x00007ffeeeb3a100
   r8: 0x0000000000000000   r9: 0x0000000000009ea4  r10: 0x00000000fffffff0  r11: 0x0000000000000000
  r12: 0x00007f8a08e864d0  r13: 0x0000000000000000  r14: 0x0000000000000000  r15: 0x00007f8a08e864d0
  rip: 0x00000001047cce6c  rfl: 0x0000000000010246  cr2: 0x0000000000000000
  
Logical CPU:     10
Error Code:      0x00000004 (no mapping for user data read)
Trap Number:     14

Thread 0 instruction stream:
  90 90 90 90 90 90 90 90-90 90 90 90 90 90 55 48  ..............UH
  89 e5 80 7f 10 52 75 15-48 8b 47 e0 48 85 c0 74  .....Ru.H.G.H..t
  0c 80 78 10 00 75 06 f6-40 21 20 75 04 31 c0 5d  ..x..u..@! u.1.]
  c3 5d e9 d3 04 09 00 90-90 90 90 90 90 90 90 90  .]..............
  90 90 90 90 55 48 89 e5-48 8b 47 38 48 85 c0 74  ....UH..H.G8H..t
  06 48 8b 40 68 eb 02 31-c0 5d c3 90 55 48 89 e5  .H.@h..1.]..UH..
 [48]8b 07 48 8b 00 5d c3-55 48 89 e5 80 67 13 7f  H..H..].UH...g..	<==
  5d c3 55 48 89 e5 48 8d-47 d8 5d c3 55 48 89 e5  ].UH..H.G.].UH..
  48 8d 47 28 5d c3 55 48-89 e5 48 85 ff 74 0f 48  H.G(].UH..H..t.H
  8b 47 38 48 85 c0 74 06-48 8b 40 68 eb 02 31 c0  .G8H..t.H.@h..1.
  5d c3 55 48 89 e5 48 89-f8 5d c3 90 55 48 89 e5  ].UH..H..]..UH..
  41 56 53 49 89 f6 48 89-fb 48 8d 77 d8 4c 89 f7  AVSI..H..H.w.L..
  
Thread 0 last branch register state not available.


Binary Images:
       0x1010c0000 -        0x106673fff +zig (0) <B21A2A51-F18B-3CD6-88DC-087304EC0383> /Users/USER/*/zig
       0x108d39000 -        0x108db4fff +libc++.1.dylib (0) <C44B9247-092E-320B-B696-43CDC941DB40> /usr/local/opt/llvm/lib/libc++.1.dylib
       0x108e13000 -        0x108e3efff +libc++abi.1.0.dylib (0) <3682E2A1-1494-345E-9AC3-1DC1457FBBD5> /usr/local/Cellar/llvm/12.0.0/lib/libc++abi.1.0.dylib
       0x109a25000 -        0x109ac0fff  dyld (851.27) <7EAA668B-F906-3BAA-A980-139BBE6E8766> /usr/lib/dyld
    0x7fff200bb000 -     0x7fff200bcfff  libsystem_blocks.dylib (79) <1C8538C3-F2BE-3F02-89AE-E7052DE66D51> /usr/lib/system/libsystem_blocks.dylib
    0x7fff200bd000 -     0x7fff200f2fff  libxpc.dylib (2038.100.48) <CFB56DA8-7462-3E8F-9CFB-20926C6A2BB5> /usr/lib/system/libxpc.dylib
    0x7fff200f3000 -     0x7fff2010afff  libsystem_trace.dylib (1277.100.21) <1AAE1F8F-9F9D-3327-8A95-3A6887CED713> /usr/lib/system/libsystem_trace.dylib
    0x7fff2010b000 -     0x7fff201a8fff  libcorecrypto.dylib (1000.100.38) <96A88875-7771-394E-A88E-389DCD02A935> /usr/lib/system/libcorecrypto.dylib
    0x7fff201a9000 -     0x7fff201d5fff  libsystem_malloc.dylib (317.100.9) <029B5632-62B4-39F6-981C-BCA99C1FBF1D> /usr/lib/system/libsystem_malloc.dylib
    0x7fff201d6000 -     0x7fff2021afff  libdispatch.dylib (1271.100.5) <6B7B23E3-2FD4-3EA2-8A89-CE06244CCA98> /usr/lib/system/libdispatch.dylib
    0x7fff2021b000 -     0x7fff20254fff  libobjc.A.dylib (824) <929E3040-4605-3C14-885B-D742EF02F2CB> /usr/lib/libobjc.A.dylib
    0x7fff20255000 -     0x7fff20257fff  libsystem_featureflags.dylib (28.60.1) <FEA91919-A5BB-3606-9445-F2077D90BF87> /usr/lib/system/libsystem_featureflags.dylib
    0x7fff20258000 -     0x7fff202e0fff  libsystem_c.dylib (1439.100.3) <DF45CDEC-6B7E-3586-94B4-F3679A762661> /usr/lib/system/libsystem_c.dylib
    0x7fff202e1000 -     0x7fff20336fff  libc++.1.dylib (905.6) <B027735F-B398-381C-84A7-606D7BBE4997> /usr/lib/libc++.1.dylib
    0x7fff20337000 -     0x7fff2034cfff  libc++abi.dylib (905.6) <22AFC7FC-2DB6-3EF0-9CC0-EFFB9B65D5E2> /usr/lib/libc++abi.dylib
    0x7fff2034d000 -     0x7fff2037cfff  libsystem_kernel.dylib (7195.101.2) <62A19DE4-50C5-3866-B5B2-43220E379C3B> /usr/lib/system/libsystem_kernel.dylib
    0x7fff2037d000 -     0x7fff20388fff  libsystem_pthread.dylib (454.100.8) <52F807B1-41A0-3D1E-AE89-115CA570863F> /usr/lib/system/libsystem_pthread.dylib
    0x7fff20389000 -     0x7fff203c4fff  libdyld.dylib (851.27) <9F95C644-D1BD-38D9-9612-6188FE9EA53C> /usr/lib/system/libdyld.dylib
    0x7fff203c5000 -     0x7fff203cefff  libsystem_platform.dylib (254.80.2) <03429519-EBEA-3549-84A6-0FD426CB7373> /usr/lib/system/libsystem_platform.dylib
    0x7fff203cf000 -     0x7fff203fafff  libsystem_info.dylib (542.40.3) <C189F0D7-A430-328D-BD7F-7EB0FA023736> /usr/lib/system/libsystem_info.dylib
    0x7fff2255f000 -     0x7fff227befff  libicucore.A.dylib (66112) <478D57C3-FFF8-35E1-A64F-8490A616AB37> /usr/lib/libicucore.A.dylib
    0x7fff227bf000 -     0x7fff227c8fff  libsystem_darwin.dylib (1439.100.3) <28AB0CBC-61F6-3A01-BCE2-A53DA1AECB0F> /usr/lib/system/libsystem_darwin.dylib
    0x7fff22bde000 -     0x7fff22be9fff  libsystem_notify.dylib (279.40.4) <1AA89769-E07F-37CD-BBCF-6DBD345862EB> /usr/lib/system/libsystem_notify.dylib
    0x7fff24b6b000 -     0x7fff24b79fff  libsystem_networkextension.dylib (1295.101.1) <33F45B5A-D346-3E7F-AB34-DFC4387E5A3C> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff24bd8000 -     0x7fff24beefff  libsystem_asl.dylib (385) <4D4E0D4F-8B40-3ACC-85E1-16375966D6CC> /usr/lib/system/libsystem_asl.dylib
    0x7fff262b2000 -     0x7fff262b9fff  libsystem_symptoms.dylib (1431.100.22) <226C631F-5380-395A-AA20-D686AEA05B9B> /usr/lib/system/libsystem_symptoms.dylib
    0x7fff271ca000 -     0x7fff272acfff  libxml2.2.dylib (34.9) <B3F9E655-AB90-3375-8262-EC76550929D2> /usr/lib/libxml2.2.dylib
    0x7fff282b5000 -     0x7fff282c5fff  libsystem_containermanager.dylib (318.100.4) <3BAF45DD-75D2-3D8A-845E-83A7058CBA08> /usr/lib/system/libsystem_containermanager.dylib
    0x7fff28fc8000 -     0x7fff28fcbfff  libsystem_configuration.dylib (1109.101.1) <49050F1E-3143-31B9-8B5C-FCCE4C6F23E5> /usr/lib/system/libsystem_configuration.dylib
    0x7fff28fcc000 -     0x7fff28fd0fff  libsystem_sandbox.dylib (1441.101.1) <24C8DF94-E258-3E24-AC96-2D1FB85F076A> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff29cc2000 -     0x7fff29cc4fff  libquarantine.dylib (119.40.2) <FABCD375-2FD7-3FF5-9A28-452D8A323D3E> /usr/lib/system/libquarantine.dylib
    0x7fff2a26d000 -     0x7fff2a271fff  libsystem_coreservices.dylib (127) <A1451528-AD1D-3E86-93A9-1340535FAF46> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff2a46e000 -     0x7fff2a480fff  libz.1.dylib (76) <45B3DDDD-FCA5-3387-98B8-38D9BFF68588> /usr/lib/libz.1.dylib
    0x7fff2a481000 -     0x7fff2a4c8fff  libsystem_m.dylib (3186.100.3) <2F75F75F-DB36-3841-988F-48A7237ED6E9> /usr/lib/system/libsystem_m.dylib
    0x7fff2a4ca000 -     0x7fff2a4cffff  libmacho.dylib (980) <D503EC4A-6B6A-39C4-8218-9F4B6147A5BF> /usr/lib/system/libmacho.dylib
    0x7fff2a4ec000 -     0x7fff2a4f7fff  libcommonCrypto.dylib (60178.100.1) <EA0C1728-C905-3A58-A285-C32ECF903BB4> /usr/lib/system/libcommonCrypto.dylib
    0x7fff2a4f8000 -     0x7fff2a502fff  libunwind.dylib (201) <32F74246-BEB8-3249-8C51-CAD863EF158E> /usr/lib/system/libunwind.dylib
    0x7fff2a503000 -     0x7fff2a50afff  liboah.dylib (203.42) <E3126D8C-F1F5-3E30-B7A2-0EB1AB0AF12D> /usr/lib/liboah.dylib
    0x7fff2a50b000 -     0x7fff2a515fff  libcopyfile.dylib (173.40.2) <DDE4BE71-76BA-3CDC-B3B0-F5E0BADF8FF2> /usr/lib/system/libcopyfile.dylib
    0x7fff2a516000 -     0x7fff2a51dfff  libcompiler_rt.dylib (102.2) <AD12DBB0-8D0B-3D71-800D-E591A3CBE0D8> /usr/lib/system/libcompiler_rt.dylib
    0x7fff2a51e000 -     0x7fff2a520fff  libsystem_collections.dylib (1439.100.3) <8B0EA86C-AFA8-3B51-916F-7ED9A42669A4> /usr/lib/system/libsystem_collections.dylib
    0x7fff2a521000 -     0x7fff2a523fff  libsystem_secinit.dylib (87.60.1) <18494A83-796C-33D5-A6D0-BD4E0C2BBFC4> /usr/lib/system/libsystem_secinit.dylib
    0x7fff2a524000 -     0x7fff2a526fff  libremovefile.dylib (49.101.1) <DC4523F8-F5FF-3837-BD58-0AA8763F8FDE> /usr/lib/system/libremovefile.dylib
    0x7fff2a527000 -     0x7fff2a527fff  libkeymgr.dylib (31) <1A66B854-350D-3064-A606-2D9BFE71C400> /usr/lib/system/libkeymgr.dylib
    0x7fff2a528000 -     0x7fff2a52ffff  libsystem_dnssd.dylib (1310.100.10) <600B1851-C1D4-3ACF-BE2A-99E7E40E6FC0> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff2a530000 -     0x7fff2a535fff  libcache.dylib (83) <ED7D7EB4-B248-33A9-9E6A-58F45EAB7602> /usr/lib/system/libcache.dylib
    0x7fff2a536000 -     0x7fff2a537fff  libSystem.B.dylib (1292.100.5) <EADFB1D1-E113-35B4-BD2D-C1E4C38010D4> /usr/lib/libSystem.B.dylib
    0x7fff2d92c000 -     0x7fff2d92cfff  liblaunch.dylib (2038.100.48) <3C5E58AF-F753-378B-8E0C-F3C9BB82E04D> /usr/lib/system/liblaunch.dylib
    0x7fff2fdb0000 -     0x7fff2fdb0fff  libsystem_product_info_filter.dylib (8.40.1) <4CCFBF85-6483-364F-A8D6-A5D2956BEC52> /usr/lib/system/libsystem_product_info_filter.dylib
    0x7fff40c0d000 -     0x7fff40c3cfff  libncurses.5.4.dylib (57) <64DA10BA-71BD-33B0-B19B-28984C171A7D> /usr/lib/libncurses.5.4.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 803
    thread_create: 15
    thread_set_state: 63994

VM Region Summary:
ReadOnly portion of Libraries: Total=624.7M resident=0K(0%) swapped_out_or_unallocated=624.7M(100%)
Writable regions: Total=2.9G written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=2.9G(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Kernel Alloc Once                    8K        1 
MALLOC                             1.9G     1552 
MALLOC guard page                   24K        5 
MALLOC_MEDIUM (reserved)         720.0M        9         reserved VM address space (unallocated)
STACK GUARD                       56.1M       17 
Stack                            264.1M       17 
VM_ALLOCATE                          8K        2 
__DATA                            1070K       51 
__DATA_CONST                      8501K       39 
__DATA_DIRTY                        67K       24 
__LINKEDIT                       530.0M        6 
__OBJC_RO                         70.1M        1 
__OBJC_RW                         2480K        2 
__TEXT                            94.8M       50 
shared memory                        4K        1 
===========                     =======  ======= 
TOTAL                              3.6G     1777 
TOTAL, minus reserved VM space     2.9G     1777 

```

**Comment by Vexu** [comment]
Fixed by 4618c41fa6ca70f06c7e65762d2f38d57b00818c (ranges are compiled as if statements)
